include/cuco/detail/utility/cuda.hpp

```diff
@@ -58,18 +57,18 @@ constexpr auto grid_size(index_type num,
  * @return The grid size that delivers the highest occupancy
  */
 template <typename Kernel>
-constexpr auto max_occupancy_grid_size(int32_t block_size,
+constexpr auto max_occupancy_grid_size(std::int32_t block_size,
                                        Kernel kernel,
                                        std::size_t dynamic_shm_size = 0)
 {
-  int32_t device = 0;
+  std::int32_t device = 0;
   CUCO_CUDA_TRY(cudaGetDevice(&device));
 
-  int32_t num_multiprocessors = -1;
+  std::int32_t num_multiprocessors = -1;
   CUCO_CUDA_TRY(
     cudaDeviceGetAttribute(&num_multiprocessors, cudaDevAttrMultiProcessorCount, device));
 
-  int32_t max_active_blocks_per_multiprocessor{};
+  std::int32_t max_active_blocks_per_multiprocessor{};
```

Comment:
The CUDA runtime API always return builtins aka `int`.

Reply:
Can we just use `auto` in these cases so we generically adapt whatever type is given to us by the CUDA API? It should also work for your `toto` example.